src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php

```diff
@@ -344,10 +344,10 @@ public function editAction(Request $request, $supplierId)
             }
         } catch (Exception $e) {
             $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
 
-            if ($e instanceof SupplierNotFoundException || $e instanceof AddressNotFoundException) {
-                return $this->redirectToRoute('admin_suppliers_index');
-            }
+        if (!isset($supplierForm) || !isset($editableSupplier)) {
+            return $this->redirectToRoute('admin_suppliers_index');
```

Comment:
maybe this redirect could be moved to `catch` block above? then you wouldn't need the `if` condition

Reply:
We want to render the form if its possible even if exception is thrown.
If i move redirect to catch and some random exception is thrown it will redirect to index instead of rendering the form. Although I wouldn't mind that, but I guess its ux based decision.